docs/windows-design.md

```diff
@@ -0,0 +1,261 @@
+
+
+### Service Traffic
+
+Kube-proxy userspace mode is configured to provide Service functions. A specific Network Adapter named
+"HNS Internal NIC" is provided to kube-proxy to configure Service addresses. The OpenFlow entries for the Service
+traffic on Windows are the same as those on Linux.
+
+The Pod-to-Service traffic is similar to Antrea on Linux. The packet from Pod to Service leaves OVS pipeline
+on the gateway interface first. kube-proxy captures the packet on the host and then chooses a valid endpoint. 
+The destination IP of the packet is changed to the endpoint, and enters the OVS pipeline again.
+
+One difference is that on Linux, the packet is using the client Pod's IP as source address when it re-enters the
+OVS pipeline, while on Windows, it uses the gateway IP as source. This is because userspace kube-proxy on Windows
+terminates the initial connection and sets up a new connection on the gateway interface to the chosen backend Pod, 
+after which the packet is forwarded back to OVS..
+
+### External Traffic
+
+The Pod-to-external traffic is SNATed in OVS using the Node's internal IP first, and then it leaves
+the OVS pipeline on the gateway interface. As IP-Forwarding on the gateway interface is enabled, the packet
+is forwarded to the OVS bridge interface and enters OVS for the second time. 
+When the packet enters OVS at the second time, the dMAC is changed to an appropriate value using the host
+network stack. And OVS outputs the packet to the uplink interface directly.
+
+The corresponding reply traffic will enter OVS on the uplink interface. It will will access the ct zone, and
+the destination IP will be translated back to the original Pod's IP within the ct context. Then it will be
+output to the Pod.
+
+<img src="/docs/assets/windows_external_traffic.svg" width="600" alt="Traffic to external">
+
+### Host Traffic
+
+In “Transparent” mode, the Antrea Agent should also process the host traffic when necessary, which includes
+packets sent from the host to external addresses, and the ones sent from the external addresses to the host. 
+
+The host traffic case is similar to the Pod-to-external traffic, as the destination IP is not in the cluster
+CIDR, but we can differentiate based on the source IP. If the packet is sent from the OVS bridge interface,
+and if it is destined to a local Pod, it is output to the uplink interface directly. Reply traffic goes through
+conntrack first, then through the rest of the OVS pipeline if the ct_mark is set. If the packet is not set, it
+is output to the Windows host on the OVS bridge interface. 
+
+
+## Future Work
+1. Support NetworkPolicy for Service traffic. Since kube-proxy in **userspace** mode will change the source
+IP in the request packet, the NetworkPolicy rules don't work as expected. As Kubernetes Service is
+planned to be implemented based on OpenFlow in Antrea ([Issue 463](https://github.com/vmware-tanzu/antrea/issues/463)),
```

Comment:
Consider rephrasing this sentence. Like:
This issue will be fixed when Antrea implements [Kubernetes Service with OVS](...).

Reply:
Done.